# QUESTION 5

Which was the pickup zone with the largest total_amount (sum of all trips) on November 18th, 2025? (1 point) <br>

a. East Harlem North <br>
b. East Harlem South <br>
c. Morningside Heights <br>
d. Forest Hills <br>


In [1]:
#download dataset green trip data
!wget https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet

--2026-01-24 16:26:29--  https://d37ci6vzurychx.cloudfront.net/trip-data/green_tripdata_2025-11.parquet
Resolving d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)... 52.85.97.194, 52.85.97.13, 52.85.97.61, ...
Connecting to d37ci6vzurychx.cloudfront.net (d37ci6vzurychx.cloudfront.net)|52.85.97.194|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1164775 (1.1M) [binary/octet-stream]
Saving to: ‘green_tripdata_2025-11.parquet’

green_tripdata_2025 100%[===================>]   1.11M  --.-KB/s    in 0.03s   

2026-01-24 16:26:29 (34.6 MB/s) - ‘green_tripdata_2025-11.parquet’ saved [1164775/1164775]



In [2]:
#download dataset zone data
!wget https://github.com/DataTalksClub/nyc-tlc-data/releases/download/misc/taxi_zone_lookup.csv

--2026-01-24 16:26:29--  https://github.com/DataTalksClub/nyc-tlc-data/releases/download/misc/taxi_zone_lookup.csv
Resolving github.com (github.com)... 140.82.116.4
Connecting to github.com (github.com)|140.82.116.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://release-assets.githubusercontent.com/github-production-release-asset/513814948/5a2cc2f5-b4cd-4584-9c62-a6ea97ed0e6a?sp=r&sv=2018-11-09&sr=b&spr=https&se=2026-01-24T17%3A27%3A23Z&rscd=attachment%3B+filename%3Dtaxi_zone_lookup.csv&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2026-01-24T16%3A26%3A29Z&ske=2026-01-24T17%3A27%3A23Z&sks=b&skv=2018-11-09&sig=tYsSiLef64M7fhrxTc5XSxhdXnuNf1SUAnTSBVTeVMo%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2OTI3MjI5MCwibmJmIjoxNzY5MjcxOTkwLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJ

In [3]:
import pandas as pd
trip_data = pd.read_parquet("green_tripdata_2025-11.parquet")
zone_data = pd.read_csv("taxi_zone_lookup.csv")

In [ ]:
#checking the data sample
trip_data.head(2)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   

   fare_amount  ...  mta_tax  tip_amount  tolls_amount  ehail_fee  \
0          7.2  ...      0.5        1.94           0.0        NaN   
1          7.2  ...      0.5        0.00           0.0        NaN   

   improvement_surcharge  total_amount  payment_type  trip_type  \
0                    1.0         11.64           1.0        1.0   
1                    1.0          9.70           2.0        1.0   

   congestion_surcharge  cbd_congestion_fee  
0                   0.0                 0.0  
1                   0.0                 0.0  

[2 rows x 21 columns]

In [8]:
#checking the data sample
zone_data.head(2)

LocationID Borough            Zone service_zone
0           1     EWR  Newark Airport          EWR
1           2  Queens     Jamaica Bay    Boro Zone

In [11]:
#JOIN two table
df_trip_zone = trip_data.merge(
    zone_data,
    left_on="PULocationID",
    right_on="LocationID",
    how="left"
)

df_trip_zone.head(3)

VendorID lpep_pickup_datetime lpep_dropoff_datetime store_and_fwd_flag  \
0         2  2025-11-01 00:34:48   2025-11-01 00:41:39                  N   
1         2  2025-11-01 00:18:52   2025-11-01 00:24:27                  N   
2         2  2025-11-01 01:03:14   2025-11-01 01:15:24                  N   

   RatecodeID  PULocationID  DOLocationID  passenger_count  trip_distance  \
0         1.0            74            42              1.0           0.74   
1         1.0            74            42              2.0           0.95   
2         1.0            83           160              1.0           2.19   

   fare_amount  ...  improvement_surcharge  total_amount  payment_type  \
0          7.2  ...                    1.0         11.64           1.0   
1          7.2  ...                    1.0          9.70           2.0   
2         13.5  ...                    1.0         21.00           1.0   

   trip_type  congestion_surcharge  cbd_congestion_fee  LocationID    Borough  \
0        1.0                   0.0                 0.0          74  Manhattan   
1        1.0                   0.0                 0.0          74  Manhattan   
2        1.0                   0.0                 0.0          83     Queens   

                Zone  service_zone  
0  East Harlem North     Boro Zone  
1  East Harlem North     Boro Zone  
2   Elmhurst/Maspeth     Boro Zone  

[3 rows x 25 columns]

In [12]:
#Filter only trip on 2025-11-18
df_day = df_trip_zone[
    df_trip_zone["lpep_pickup_datetime"].dt.date == pd.to_datetime("2025-11-18").date()
]

#Grouping the trip on 2025-11-18 by Zone
zone_group = df_day.groupby("Zone")["total_amount"].sum().reset_index()

In [14]:
#Sorting the total_amount
zone_group.sort_values(by = 'total_amount', ascending = False)

Zone  total_amount
39          East Harlem North       9281.92
40          East Harlem South       6696.13
23               Central Park       2378.79
128  Washington Heights South       2139.05
85        Morningside Heights       2100.59
..                        ...           ...
125     Upper West Side South         16.00
124     Upper West Side North         16.00
129              West Village         16.00
12                  Bronxdale         13.57
14         Brooklyn Navy Yard          0.00

[136 rows x 2 columns]

The greatest total_amount is 9281.92, which is located at East Harlem North, <br>
so the answer is <b> East Harlem North (a) </b>